### PR TITLE
Fix #3611: Serialize null file names in StackTraceElement as "".

### LINKED ISSUE #3611: Test failures are ignored when filename is null in the stack trace
Example test code (based on scala-js/scalajs-tutorial):
```
package tutorial.webapp

import utest._
import scala.scalajs.js

@js.native
trait Foo extends js.Object {
 var x: js.Dictionary[Int] = js.native
}

object TutorialTest extends TestSuite {

  def tests = Tests {
    'HelloWorld - {
      val foo = js.Dynamic.literal(x = null).asInstanceOf[Foo]
      assert(foo.x.values.exists(_ > 0))
    }
  }
}
```

and the result:

```
sbt:Scala.js Tutorial> test
-------------------------------- Running Tests --------------------------------
X tutorial.webapp.TutorialTest.HelloWorld 4ms
  utest.AssertionError: assert(foo.x.values.exists(_ > 0))
  foo: tutorial.webapp.Foo = [object Object]
  scala.scalajs.js.JavaScriptException: TypeError: Cannot convert undefined or null to object
    <jscode>.Function.keys@<anonymous>(Unknown)
    scala.scalajs.js.WrappedDictionary$DictionaryIterator.<init>(R:\scalajs-tutorial-0.6.x\target\sc
    ala-2.12\scala-js-tutorial-test-fastopt.js:23811)
    scala.scalajs.js.WrappedDictionary.iterator(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js
    -tutorial-test-fastopt.js:46729)
    scala.collection.MapLike$$anon$2.<init>(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tut
    orial-test-fastopt.js:26890)
    <jscode>.$c_sjs_js_WrappedDictionary.$c_sc_AbstractMap.valuesIterator__sc_Iterator(R:\scalajs-tu
    torial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:40509)
    <jscode>.{anonymous}()(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastop
    t.js:17605)
    scala.scalajs.runtime.AnonFunction1.apply(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-t
    utorial-test-fastopt.js:20908)
    utest.asserts.Util$.runAssertionEntry(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutor
    ial-test-fastopt.js:6632)
[error] scala.scalajs.js.JavaScriptException: TypeError: Cannot read property 'length' of null
[error]         at java.io.DataOutputStream.writeUTF(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:31125)
[error]         at org.scalajs.testcommon.Serializer$StackTraceElementSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13354)
[error]         at org.scalajs.testcommon.Serializer$StackTraceElementSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13346)
[error]         at org.scalajs.testcommon.Serializer$$anon$4.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:12834)
[error]         at org.scalajs.testcommon.Serializer$ThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13570)
[error]         at org.scalajs.testcommon.Serializer$ThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13577)
[error]         at org.scalajs.testcommon.Serializer$$anon$5.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:12887)
[error]         at org.scalajs.testcommon.Serializer$ThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13574)
[error]         at org.scalajs.testcommon.Serializer$OptionalThrowableSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:13199)
[error]         at org.scalajs.testcommon.Serializer$EventSerializer$.serialize(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:12985)
    <jscode>.{anonymous}()(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastop
    t.js:17635)
    scala.scalajs.runtime.AnonFunction0.apply(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-t
    utorial-test-fastopt.js:20880)
[info]
[success] Total time: 5 s, completed 2019/04/08 20:58:04
```

Example 2:

```
package tutorial.webapp

import utest._
import scala.scalajs.js

object TutorialTest extends TestSuite {

  def tests = Tests {
    'HelloWorld - {
      js.Dynamic.global.eval("throw new Error()")
      assert(true)
    }
  }
}
```

and the result:

```
...
-------------------------------- Running Tests --------------------------------
X tutorial.webapp.TutorialTest.HelloWorld 1ms
  scala.scalajs.js.JavaScriptException: Error
    <jscode>.eval(eval at <anonymous> (R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial
    -test-fastopt.js:17595:21), <anonymous>:1)
    <jscode>.eval@<anonymous>(Unknown)
    <jscode>.{anonymous}()(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastop
    t.js:17595)
    scala.scalajs.runtime.AnonFunction0.apply(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-t
    utorial-test-fastopt.js:20868)
[error] scala.scalajs.js.JavaScriptException: TypeError: Cannot read property 'length' of null
[error]         at java.io.DataOutputStream.writeUTF(R:\scalajs-tutorial-0.6.x\target\scala-2.12\scala-js-tutorial-test-fastopt.js:30907)
...
```

Although utest clearly prints about test failure, `test` task was finished as `[success]` and  TestsFailedException hasn't be thrown.

I think this JavaScriptException has a StackTraceElement with filename=null, and the filename (null) has been passed to DataOutputStream.writeUTF which doesn't expect null.

sbt 1.2.8 / scala 2.12.6 / sbt-scalajs 0.6.25 / utest 0.6.3

### PR DESCRIPTION
**Pull request by exoego**
Closes #3611 

I wonder where test for this should be added
`test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/StackTraceElementTest.scala` is right place?

### CODE REVIEW

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/StackTraceElementTest.scala [hunk 1]
```diff
@@ -14,11 +14,26 @@ package org.scalajs.testsuite.javalib.lang
 
 import org.junit.Assert._
 import org.junit.Test
+import java.io._
+
+import org.scalajs.testcommon._
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/StackTraceElementTest.scala]: This doesn't compile when doing `testSuiteJVM/test`. The correct place for tests of the test-common serializers is in https://github.com/scala-js/scala-js/blob/ccdc1f00d100a151457aabc86d1d7f05dafd9063/test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala#L18.

Comment (exoego) [reply test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/StackTraceElementTest.scala]: Fixed in https://github.com/scala-js/scala-js/pull/3631/commits/31bad02559e0a73cbf9e0cd486f1a6b3a235a011

**Comment by exoego** [comment]
> 
> 
> I think we should also adapt the deserializer to turn back `""` into `null`. That would provide a better experience when the deserializer is recent enough.

Fixed in https://github.com/scala-js/scala-js/pull/3631/commits/107b181f972e18089759e284f945007b352f67d0
Confirmed by `testAdapter/test`

**Comment by gzm0** [comment]
I think this commit should be [no-master]? Or at least a comment that we should properly serialize it as option in master where we can break interface compatibility?

File: test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala [hunk 2]
```diff
@@ -28,4 +30,12 @@ class SerializerTest {
     assertEquals(in.toString(), out.toString())
     assertEquals(in.getStackTrace().size, out.getStackTrace().size)
   }
+
+  @Test
+  def serializeAndDesirializeIfFileNameIsNull: Unit = {
```
Comment (gzm0) [inline test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala]: Two things:
- Name: `serializeStackTraceElementWithNullFilename` (mainly for consistency with the other test).
- Reference #3611 (just add `// # 3611`).

Comment (exoego) [reply test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala]: Addressed in https://github.com/scala-js/scala-js/pull/3631/commits/8cc42399ebe2655f0b7a45bcdb7415eb734b920f and https://github.com/scala-js/scala-js/pull/3631/commits/169adb6c0544a9e0587d096bdd07abc1d6eaf87a

File: test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala [hunk 3]
```diff
@@ -12,6 +12,8 @@
 
 package org.scalajs.testcommon
 
+import java.io._
```
Comment (sjrd) [inline test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala]: Unused import?

**Comment by sjrd** [comment]
> I think this commit should be [no-master]? Or at least a comment that we should properly serialize it as option in master where we can break interface compatibility?

Filed as #3634.

File: test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala (hunk 3 above)
Comment (exoego) [reply test-common/src/test/scala/org/scalajs/testcommon/SerializerTest.scala]: Removed unused import.

And squashed commits into https://github.com/scala-js/scala-js/pull/3631/commits/1aaf291a4fe323303a69c15e55437e0b0ef3a264

**Pull request merged**